kmir/src/kmir/kmir.py

```diff
@@ -62,127 +58,15 @@ def __init__(
-                    str(llvm_def_output),
-                    str(target_llvmdt_path),
-                    'c',
-                    '-O2',
-                    '--',
-                    '-o',
-                    target_llvm_path / 'interpreter',
-                ],
-                check=True,
-            )
-
-            # Process Haskell definition
-            _LOGGER.info('Writing Haskell definition file')
-            hs_def_file = HASKELL_DEF_DIR / 'definition.kore'
-            _insert_rules_and_write(hs_def_file, rules, target_hs_path / 'definition.kore')
-
-            # Copy all files except definition.kore and binary from HASKELL_DEF_DIR to out/hs
-            _LOGGER.info('Copying other artefacts into HS output directory')
-            for file_path in HASKELL_DEF_DIR.iterdir():
-                if file_path.name != 'definition.kore' and file_path.name != 'haskellDefinition.bin':
-                    if file_path.is_file():
-                        shutil.copy2(file_path, target_hs_path / file_path.name)
-                    elif file_path.is_dir():
-                        shutil.copytree(file_path, target_hs_path / file_path.name, dirs_exist_ok=True)
-            return KMIR(target_hs_path, target_llvm_path, bug_report=bug_report)
-        else:
-
-            target_llvm_path = target_path / 'llvm'
-            target_llvmdt_path = target_llvm_path / 'dt'
-            _LOGGER.info(f'Creating directory {target_llvmdt_path}')
-            target_llvmdt_path.mkdir(parents=True, exist_ok=True)
-
-            # Process LLVM definition
-            _LOGGER.info('Writing LLVM definition file')
-            llvm_def_file = LLVM_LIB_DIR / 'definition.kore'
-            llvm_def_output = target_llvm_path / 'definition.kore'
-            _insert_rules_and_write(llvm_def_file, rules, llvm_def_output)
-
-            import subprocess
-
-            _LOGGER.info('Running llvm-kompile-matching')
-            subprocess.run(
-                ['llvm-kompile-matching', str(llvm_def_output), 'qbaL', str(target_llvmdt_path), '1/2'], check=True
-            )
-            _LOGGER.info('Running llvm-kompile')
-            subprocess.run(
-                [
-                    'llvm-kompile',
-                    str(llvm_def_output),
-                    str(target_llvmdt_path),
-                    'main',
-                    '-O2',
-                    '--',
-                    '-o',
-                    target_llvm_path / 'interpreter',
-                ],
-                check=True,
-            )
-            blacklist = ['definition.kore', 'interpreter', 'dt']
-            to_copy = [file.name for file in LLVM_DEF_DIR.iterdir() if file.name not in blacklist]
-            for file in to_copy:
-                _LOGGER.info(f'Copying file {file}')
-                shutil.copy2(LLVM_DEF_DIR / file, target_llvm_path / file)
-            return KMIR(target_llvm_path, None, bug_report=bug_report)
+        from .kompile import kompile_smir
```

Comment:
This creates an import loop (however only in local scope), that's why I didn't continue with factoring out the compilation code. If that's not a problem I have no objections.